helix-term/src/application.rs

```diff
@@ -1134,6 +1134,84 @@ impl Application {
                         }
                         Ok(serde_json::Value::Null)
                     }
+                    Ok(MethodCall::ShowDocument(params)) => {
+                        let language_server = language_server!();
+                        let offset_encoding = language_server.offset_encoding();
+
+                        let success = match params {
```

Comment:
a nested match/if assignment like this is pretty verbose. I would just use `let mut success = false` and only set it to true if the command succeded (two places).

Also using a match is pretty verbose here. I would just use early return liberally  (`return Ok(json!(lsp::ShowDocumentResult { success: true/false }))` is simple enough to repeat a few times).

Reply:
Ended up extracting the handling logic into separate method which ended up being way cleaner.

Reply:
Yeah that is better